src/test/java/com/stripe/model/EphemeralKeyTest.java

```diff
@@ -1,104 +1,26 @@
-import java.util.Map;
-
-import org.junit.After;
-import org.junit.Before;
 import org.junit.Test;
 
 public class EphemeralKeyTest extends BaseStripeTest {
-  private String oldApiVersion;
-
-  @Before
-  public void mockStripeResponseGetter() {
-    APIResource.setStripeResponseGetter(networkMock);
-  }
-
-  @After
-  public void unmockStripeResponseGetter() {
-    /* This needs to be done because tests aren't isolated in Java */
-    APIResource.setStripeResponseGetter(new LiveStripeResponseGetter());
-  }
-
-  @Before
-  public void saveOldStripeVersion() {
-    oldApiVersion = Stripe.apiVersion;
-  }
-
-  @After
-  public void restoreOldStripeVersion() {
-    Stripe.apiVersion = oldApiVersion;
-  }
-
-  @Test
-  public void testCreate() throws StripeException {
-    Stripe.apiVersion = "2016-06-05";
-
-    final Map<String, Object> params = new HashMap<String, Object>();
-    params.put("customer", "cus_123");
-
-    final RequestOptions options = RequestOptions.builder().setStripeVersion("2017-05-25").build();
-
-    EphemeralKey.create(params, options);
-
-    verifyPost(EphemeralKey.class, "https://api.stripe.com/v1/ephemeral_keys", params, options);
-    verifyNoMoreInteractions(networkMock);
-  }
-
-  @Test(expected = IllegalArgumentException.class)
-  public void testVersionlessCreate() throws StripeException {
-    Stripe.apiVersion = null;
-
-    final Map<String, Object> params = new HashMap<String, Object>();
-    params.put("customer", "cus_123");
-
-    final RequestOptions options = RequestOptions.getDefault();
-
-    EphemeralKey.create(params, options);
-  }
-
   @Test
-  public void testDelete() throws StripeException {
-    EphemeralKey key = new EphemeralKey();
-    key.setId("ephkey_123");
-
-    key.delete();
-
-    verifyDelete(EphemeralKey.class, "https://api.stripe.com/v1/ephemeral_keys/ephkey_123");
-    verifyNoMoreInteractions(networkMock);
-  }
-
-  @Test
-  public void testDeserialize() throws StripeException, IOException {
-    final String json = resource("ephemeral_key.json");
-    EphemeralKey key = APIResource.GSON.fromJson(json, EphemeralKey.class);
-
-    assertEquals(key.getId(), "ephkey_123");
-    assertEquals(key.getObject(), "ephemeral_key");
-    assertEquals(key.getCreated(), Long.valueOf(100));
-    assertEquals(key.getExpires(), Long.valueOf(200));
-    assertEquals(key.getLivemode(), false);
-    assertEquals(key.getSecret(), "ek_test_hunter2");
-    assertEquals(key.getAssociatedObjects().size(), 1);
-    assertEquals(key.getAssociatedObjects().get(0).getType(), "customer");
-    assertEquals(key.getAssociatedObjects().get(0).getId(), "cus_234");
+  public void testDeserialize() throws Exception {
+    String json = getResourceAsString("/api_fixtures/ephemeral_key.json");
+    EphemeralKey resource = APIResource.GSON.fromJson(json, EphemeralKey.class);
+    assertNotNull(resource);
+    assertEquals("ephkey_123", resource.getId());
+    assertEquals("ephemeral_key", resource.getObject());
```

Comment:
why not use stripe-mock?

Reply:
Because ephemeral keys can only be created, not retrieved, and require special parameters for the API version. Since this test is specifically for the deserialization, I didn't want to introduce additional complexity by using stripe-mock here.